R/nested.R

```diff
@@ -88,3 +88,49 @@ create_filler_nested <- function(pd_nested) {
   select_(pd_nested, ~spaces, ~newlines, ~short, ~everything())
 }
 
+#' Serialize a nested parse table
+#'
+#' Helper function that recursively extracts terminals from a nested tibble.
+#' @param pd_nested A nested parse table.
+#' @param pass_indent Level of indention of a token.
+#' @return A character vector with all tokens in `pd_nested` plus whitespace,
+#'   line break and indention information added between the tokens.
+#' @importFrom purrr pmap
+serialize_parse_data_nested_helper <- function(pd_nested, pass_indent) {
+  out <- pmap(list(pd_nested$terminal, pd_nested$text, pd_nested$child,
+             pd_nested$spaces, pd_nested$newlines, pd_nested$indent),
+           function(terminal, text, child, spaces, newlines, indent) {
+             if (terminal) {
+               c(rep_char(" ", pass_indent), text,
```

Comment:
Yes, in general I think it can, but if this if clause applies, we are in a terminal token, for which text cannot be newlines ("\n") I think. It could be a string like "'\n'" though, but then it would remain a string and not get evaluated.